src/http/http_response.f90

```diff
@@ -36,6 +39,24 @@ subroutine append_header(this, key, value)
         end if
     
     end subroutine append_header
-  
 
+    pure function header_value(this, key) result(retval)
```

Comment:
Can you add a short docstring that describes what this does? Mention the state of the result value if the desired key is not found, as well as the behavior that we return value of the first `key` that is found, in case there are duplicates.